test/LRPTests/APITests.cpp

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include <TestDef.h>
+#include <TlHelp32.h>
+#include <NotificationsLongRunningProcess_h.h>
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::System;
+
+namespace Test::LRP
+{
+    static const PCWSTR c_processName = L"TAEF.exe";
+    static const PCWSTR c_appId = L"toastAppId";
+
+    class LRPTests
+    {
+
+    public:
+        BEGIN_TEST_CLASS(LRPTests)
+            TEST_CLASS_PROPERTY(L"Description", L"Windows App SDK Push Notifications Long Running Process tests")
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunAs:Class", L"RestrictedUser")
+        END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages(Test::Bootstrap::Packages::Singleton);
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            ::Test::Bootstrap::CleanupPackages(Test::Bootstrap::Packages::Singleton);
+            return true;
+        }
+
+        TEST_METHOD(LaunchLRP_FromCoCreateInstance)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            VERIFY_IS_NOT_NULL(notificationPlatform.get());
+        }
+
+        TEST_METHOD(LaunchLRP_FromStartupTask)
+        {
+            STARTUPINFO startupInfo = { 0 };
+            wil::unique_process_information processInfo;
+
+            ZeroMemory(&startupInfo, sizeof(startupInfo));
+            startupInfo.cb = sizeof(startupInfo);
+
+            // Build the Solution OutDir path where the Startup Task exe is.
+            auto startupTaskExePath = Test::FileSystem::GetSolutionOutDirPath();
+            startupTaskExePath /= Test::Packages::WindowsAppRuntimeSingleton::c_PackageDirName;
+            startupTaskExePath += L".Msix";
+            startupTaskExePath /= L"msix";
+            startupTaskExePath /= L"PushNotificationsLongRunningTask.StartupTask.exe";
+
+            BOOL wasProcessCreated = CreateProcess(
+                startupTaskExePath.c_str(),
+                nullptr, // command line options
+                nullptr, // process attributes
+                nullptr, // thread attributes
+                FALSE,   // inherit handles
+                NORMAL_PRIORITY_CLASS, // creation flags
+                nullptr, // lpEnvironment
+                nullptr, // current directory for the process
+                &startupInfo,
+                &processInfo
+            );
+
+            if (wasProcessCreated == FALSE)
```

Comment:
Why do we need this check? Shouldn't we fail the test if process creation failed?

Reply:
If wasProcessCreated is false, GetLastError will contain the error code and the next line uses VERIFY_SUCCEEDED to log the exact error that was thrown.

Reply:
use winrt for getlasterror

Reply:
winrt::check_bool will throw lasterror I think.

Reply:
Yup. https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/error-handling/check-bool